internal/backend/service.go

```diff
@@ -29,11 +84,114 @@ func NewDefaultService(c Persister, l log.Logger) *DefaultService {
 	}
 }
 
-// GetAccessRequest will retrieve the access request from k8s identified by the
-// given name and namespace. Will return a nil value without any error if the
-// access request isn't found.
-func (s *DefaultService) GetAccessRequest(ctx context.Context, name, namespace string) (*api.AccessRequest, error) {
-	ar, err := s.k8s.GetAccessRequest(ctx, name, namespace)
+// GetAccessRequest will retrieve the access request for the specified role.
+// Will return a nil value without any error if an access request isn't found for this role.
+func (s *DefaultService) GetAccessRequest(ctx context.Context, key *AccessRequestKey, roleName string) (*api.AccessRequest, error) {
+
+	// get all access requests
+	accessRequests, err := s.ListAccessRequests(ctx, key, true)
+	if err != nil {
+		return nil, fmt.Errorf("error listing access request for role %s: %w", roleName, err)
+	}
+
+	// find the first access request matching the requested role
+	for _, ar := range accessRequests {
+		if ar.Spec.Role.TemplateName == roleName {
+			return ar, nil
+		}
+	}
```

Comment:
I think this logic belongs to the `ListAccessRequests`. We could create filter struct an pass it to the `ListAccessRequests`. Something similar to the kubernetes client.ListOptions.

Reply:
Could be a good refactor. But since it is currently the only "list option", and get will need to iterate to get the first item anyway as list should return a slice, I think it can be implemented later when it is worth the added complexity